### ISSUE TITLE
MCF reader time axis validation

### ISSUE THREAD

**Issue by taobrienlbl** [LBL-EESA/TECA#635]
I discovered an issue with TECA BARD, where two variables lose temporal coherence with time (relative to the simulation).

TECA BARD was run on output from a CAM5 C20C+ simulation with the following command (I'm aware that the `--steps_per_file` option isn't doing what I expected...the script that generated this run command was written before the changes to the output file layout, and I forgot to update that):

```bash
srun -N 20 -n 340 \
    teca_bayesian_ar_detect \
        --input_file mcf_files_v3/All-Hist_est1_v2-0_run001.mcf \
        --specific_humidity hus --wind_u ua --wind_v va \
        --ivt_u uhusavi --ivt_v vhusavi --ivt windhusavi \
        --compute_ivt --write_ivt --write_ivt_magnitude --steps_per_file 2920 \
        --ar_weighted_variables pr --output_file C20C_BARD_outputs_v3/LBNL/CAM5-1-1degree/All-Hist/est1/v2-0/3hr/atmos/windhusavi/run001//windhusavi_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_%t%.nc
```

The MCF file used is shown below:

```python
!cat /global/cscratch1/sd/taobrien/C20C-AR-Analysis/mcf_files_v3/All-Hist_est1_v2-0_run001.mcf
```

    # TECA multi cf reader config
    
    data_root = /global/cscratch1/sd/taobrien/C20C_ua_va_hus/linked_inputs/LBNL/CAM5-1-1degree/All-Hist/est1/v2-0/3hr/atmos/
    regex = A3hr_CAM5-1-1degree_All-Hist_est1_v2-0.*\.nc$
    
    z_axis_variable = plev
    
    [cf_reader]
    variables = hus
    regex = %data_root%/hus/run001/hus_%regex%
    provides_time
    provides_geometry
    
    [cf_reader]
    variables = ua
    regex = %data_root%/ua/run001/ua_%regex%
    
    [cf_reader]
    variables = va
    regex = %data_root%/va/run001/va_%regex%
    
    [cf_reader]
    variables = pr
    regex = %data_root%/pr/run001/pr_%regex%
    z_axis_variable = height
    clamp_dimensions_of_one = 1

In the initial part of the run, `windhusavi` and `pr` have a high degree of spatial correlation, with high precipitation often corresponding to areas with high values of IVT, which is expected:

```python
import matplotlib as mpl
import pylab as PP
import cartopy
import xarray as xr
mpl.style.use('seaborn-poster')

input_file = "/global/cscratch1/sd/taobrien/C20C-AR-Analysis/C20C_BARD_outputs_v3/LBNL/CAM5-1-1degree/All-Hist/est1/v2-0/3hr/atmos/windhusavi/run001/windhusavi_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_1959-01-01-01Z.nc"
with xr.open_dataset(input_file) as fin:
    for itime in [0, -1]:
        windhusavi = fin['windhusavi'].isel(time = itime)
        pr = fin['pr'].isel(time = itime)

        fig, ax = PP.subplots(1,1,subplot_kw = dict(projection = cartopy.crs.PlateCarree()))

        windhusavi.plot.contour(transform = cartopy.crs.PlateCarree(), cmap = 'viridis', alpha = 0.5, linewidths = 1, add_colorbar = True)
        pr.plot(transform = cartopy.crs.PlateCarree(), vmin = 0, cmap = 'magma')

        ax.coastlines(color = 'white', alpha = 0.5)

        PP.show()

```

    
![output_3_0](https://user-images.githubusercontent.com/8796694/127669725-03bc5ec3-c796-4710-9164-620ba2444362.png)
    

    
![output_3_1](https://user-images.githubusercontent.com/8796694/127669727-9088ff2d-c1ea-487e-b694-1bcdd5113e39.png)
    

However, later in the simulation, the spatial correlation is not as high:

```python
input_file = "/global/cscratch1/sd/taobrien/C20C-AR-Analysis/C20C_BARD_outputs_v3/LBNL/CAM5-1-1degree/All-Hist/est1/v2-0/3hr/atmos/windhusavi/run001/windhusavi_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_2017-01-01-01Z.nc"
with xr.open_dataset(input_file) as fin:
    for itime in [0]:
        windhusavi = fin['windhusavi'].isel(time = itime)
        pr = fin['pr'].isel(time = itime)

        fig, ax = PP.subplots(1,1,subplot_kw = dict(projection = cartopy.crs.PlateCarree()))

        windhusavi.plot.contour(transform = cartopy.crs.PlateCarree(), cmap = 'viridis', alpha = 0.5, linewidths = 1, add_colorbar = True)
        pr.plot(transform = cartopy.crs.PlateCarree(), vmin = 0, cmap = 'magma')

        ax.coastlines(color = 'white', alpha = 0.5)

        PP.show()
```

    
![output_5_0](https://user-images.githubusercontent.com/8796694/127669730-ee83b477-4ccf-4039-9945-3ea37875adca.png)
    

It almost looks as though precipitation is lagging IVT by ~1 day. Note, for example, how the precipitation maximum near the west coast of N. America is to the northwest of the IVT maximum with the associated AR.

This reduction in spatial correlation is systematic with time.  I wrote a TECA algorithm (`/global/project/projectdirs/m1517/cascade/taobrien/c20c_ars/ar_precip_check/teca_get_correlations.py`) to calculate the spatial correlation coefficient between these two variables at all timesteps:

```python
with xr.open_dataset("correlations_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_000000.nc") as fin:
    fin['correlation'].plot();
```

    
![output_7_0](https://user-images.githubusercontent.com/8796694/127669731-a679031d-1bc5-4ec3-8240-16a4c0cecf1a.png)
    

There's no physical reason to expect this reduction in temporal coherence with time (albeit there might be some varition within a given year, associated with the annual cycle of atmospheric rivers)

I double-checked that all the input files in the run have the same timesteps by running `teca_metadata_probe` on the regexes for each of the four variables: `hus`, `ua`, `va`, and `pr`.

TECA finds the same exact time information, including # of timesteps, for all four groups of files:
```A total of 169334 steps available in 61 files. Using the 365_day calendar.
Times are specified in units of days since 1958-1-1 00:00:00. The available
times range from 1959-1-1 1:30:0 (365.062) to 2018-12-31 22:30:0 (22264.9).
```

The full output of the four `teca_metadata_probe` runs follows:

```python
!cat metadata_probe_report_new_files_run001.txt
```

    
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 52 "ua_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201301010000-201312312359.nc" units "days since 2013-1-1 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 53 "ua_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201401010000-201409302359.nc" units "days since 2013-1-1 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 54 "ua_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201410010000-201412312359.nc" units "days since 2014-10-01 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 55 "ua_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201501010000-201512312359.nc" units "days since 2014-10-01 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 56 "ua_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201601010000-201612312359.nc" units "days since 2014-10-01 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 57 "ua_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201701010000-201706302359.nc" units "days since 2017-01-01 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 58 "ua_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201707010000-201712312359.nc" units "days since 2017-7-1 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 59 "ua_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201801010000-201802282359.nc" units "days since 2017-7-1 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 60 "ua_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201803010000-201812312359.nc" units "days since 2017-7-1 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    
    A total of 169334 steps available in 61 files. Using the 365_day calendar.
    Times are specified in units of days since 1958-1-1 00:00:00. The available
    times range from 1959-1-1 1:30:0 (365.062) to 2018-12-31 22:30:0 (22264.9).
    
    Mesh dimension: 2D
    Mesh coordinates: lon, lat
    Mesh extents: 0, 287, 0, 191
    Mesh bounds: 0, 358.75, -90, 90
    
    6 data arrays available
    
      Id   Name         Type        Centering     Dimensions                Shape                    
    -----------------------------------------------------------------------------------------------
      1    lat          NC_FLOAT    point 1D      [lat]                     [192]                    
      2    lon          NC_FLOAT    point 1D      [lon]                     [288]                    
      3    plev         NC_FLOAT    none 0D       [plev]                    [8]                      
      4    time         NC_FLOAT    point 0D      [time]                    [169334]                 
      5    time_bnds    NC_FLOAT    none 0D       [time, bnds]              [169334, 2]              
      6    ua           NC_FLOAT    none 2D       [time, plev, lat, lon]    [169334, 8, 192, 288]    
    
    srun: Job 44924034 step creation temporarily disabled, retrying (Requested nodes are busy)
    srun: Step created for job 44924034
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 52 "va_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201301010000-201312312359.nc" units "days since 2013-1-1 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 53 "va_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201401010000-201409302359.nc" units "days since 2013-1-1 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 54 "va_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201410010000-201412312359.nc" units "days since 2014-10-01 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 55 "va_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201501010000-201512312359.nc" units "days since 2014-10-01 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 56 "va_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201601010000-201612312359.nc" units "days since 2014-10-01 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 57 "va_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201701010000-201706302359.nc" units "days since 2017-01-01 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 58 "va_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201707010000-201712312359.nc" units "days since 2017-7-1 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 59 "va_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201801010000-201802282359.nc" units "days since 2017-7-1 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 60 "va_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201803010000-201812312359.nc" units "days since 2017-7-1 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    
    A total of 169334 steps available in 61 files. Using the 365_day calendar.
    Times are specified in units of days since 1958-1-1 00:00:00. The available
    times range from 1959-1-1 1:30:0 (365.062) to 2018-12-31 22:30:0 (22264.9).
    
    Mesh dimension: 2D
    Mesh coordinates: lon, lat
    Mesh extents: 0, 287, 0, 191
    Mesh bounds: 0, 358.75, -90, 90
    
    6 data arrays available
    
      Id   Name         Type        Centering     Dimensions                Shape                    
    -----------------------------------------------------------------------------------------------
      1    lat          NC_FLOAT    point 1D      [lat]                     [192]                    
      2    lon          NC_FLOAT    point 1D      [lon]                     [288]                    
      3    plev         NC_FLOAT    none 0D       [plev]                    [8]                      
      4    time         NC_FLOAT    point 0D      [time]                    [169334]                 
      5    time_bnds    NC_FLOAT    none 0D       [time, bnds]              [169334, 2]              
      6    va           NC_FLOAT    none 2D       [time, plev, lat, lon]    [169334, 8, 192, 288]    
    
    srun: Job 44924034 step creation temporarily disabled, retrying (Requested nodes are busy)
    srun: Step created for job 44924034
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 52 "hus_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201301010000-201312312359.nc" units "days since 2013-1-1 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 53 "hus_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201401010000-201409302359.nc" units "days since 2013-1-1 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 54 "hus_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201410010000-201412312359.nc" units "days since 2014-10-01 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 55 "hus_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201501010000-201512312359.nc" units "days since 2014-10-01 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 56 "hus_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201601010000-201612312359.nc" units "days since 2014-10-01 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 57 "hus_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201701010000-201706302359.nc" units "days since 2017-01-01 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 58 "hus_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201707010000-201712312359.nc" units "days since 2017-7-1 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 59 "hus_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201801010000-201802282359.nc" units "days since 2017-7-1 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 60 "hus_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201803010000-201812312359.nc" units "days since 2017-7-1 00:00:00" differs from base units "days since 1958-1-1 00:00:00" a conversion will be made.
    
    A total of 169334 steps available in 61 files. Using the 365_day calendar.
    Times are specified in units of days since 1958-1-1 00:00:00. The available
    times range from 1959-1-1 1:30:0 (365.062) to 2018-12-31 22:30:0 (22264.9).
    
    Mesh dimension: 2D
    Mesh coordinates: lon, lat
    Mesh extents: 0, 287, 0, 191
    Mesh bounds: 0, 358.75, -90, 90
    
    6 data arrays available
    
      Id   Name         Type        Centering     Dimensions                Shape                    
    -----------------------------------------------------------------------------------------------
      1    hus          NC_FLOAT    none 2D       [time, plev, lat, lon]    [169334, 8, 192, 288]    
      2    lat          NC_FLOAT    point 1D      [lat]                     [192]                    
      3    lon          NC_FLOAT    point 1D      [lon]                     [288]                    
      4    plev         NC_FLOAT    none 0D       [plev]                    [8]                      
      5    time         NC_FLOAT    point 0D      [time]                    [169334]                 
      6    time_bnds    NC_FLOAT    none 0D       [time, bnds]              [169334, 2]              
    
    srun: Job 44924034 step creation temporarily disabled, retrying (Requested nodes are busy)
    srun: Step created for job 44924034
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 52 "pr_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201301010000-201312312359.nc" units "days since 2013-01-01 00:00:00" differs from base units "days since 1958-01-01 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 53 "pr_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201401010000-201409302359.nc" units "days since 2013-01-01 00:00:00" differs from base units "days since 1958-01-01 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 54 "pr_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201410010000-201412312359.nc" units "days since 2014-10-01 00:00:00" differs from base units "days since 1958-01-01 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 55 "pr_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201501010000-201512312359.nc" units "days since 2014-10-01 00:00:00" differs from base units "days since 1958-01-01 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 56 "pr_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201601010000-201612312359.nc" units "days since 2014-10-01 00:00:00" differs from base units "days since 1958-01-01 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 57 "pr_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201701010000-201706302359.nc" units "days since 2017-01-01 00:00:00" differs from base units "days since 1958-01-01 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 58 "pr_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201707010000-201712312359.nc" units "days since 2017-7-1 00:00:00" differs from base units "days since 1958-01-01 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 59 "pr_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201801010000-201802282359.nc" units "days since 2017-7-1 00:00:00" differs from base units "days since 1958-01-01 00:00:00" a conversion will be made.
    WARNING: [0:46912496728960] [/global/cscratch1/sd/loring/teca_testing/TECA_superbuild/build-develop-8e6c7571/TECA-prefix/src/TECA/io/teca_cf_reader.cxx:655 4.0.0-230-g8e6c757]
    WARNING: File 60 "pr_A3hr_CAM5-1-1degree_All-Hist_est1_v2-0_run001_201803010000-201812312359.nc" units "days since 2017-7-1 00:00:00" differs from base units "days since 1958-01-01 00:00:00" a conversion will be made.
    
    A total of 169354 steps available in 61 files. Using the 365_day calendar.
    Times are specified in units of days since 1958-01-01 00:00:00. The available
    times range from 1959-1-1 1:30:0 (365.062) to 2018-12-31 22:30:0 (22264.9).
    
    Mesh dimension: 2D
    Mesh coordinates: lon, lat
    Mesh extents: 0, 287, 0, 191
    Mesh bounds: 0, 358.75, -90, 90
    
    6 data arrays available
    
      Id   Name         Type        Centering     Dimensions                  Shape                    
    -------------------------------------------------------------------------------------------------
      1    height       NC_FLOAT    none 0D       [height]                    [1]                      
      2    lat          NC_FLOAT    point 1D      [lat]                       [192]                    
      3    lon          NC_FLOAT    point 1D      [lon]                       [288]                    
      4    pr           NC_FLOAT    none 2D       [time, height, lat, lon]    [169354, 1, 192, 288]    
      5    time         NC_FLOAT    point 0D      [time]                      [169354]                 
      6    time_bnds    NC_FLOAT    none 0D       [time, nb2]                 [169354, 2]

**Comment by taobrienlbl** [comment]
Talking with Yang about this made me realize there could be a problem with the underlying dataset.  Let me investigate this first.

**Comment by burlen** [comment]
I will add some error checking in the mcf reader to ensure consistency in the time dimension.

**Comment by taobrienlbl** [comment]
@burlen -- looks like we both overlooked something important in the `teca_metadata_probe` outputs.  The variables `ua`, `va`, and `hus` all have 169334 timesteps, whereas the `pr` files have 1693**5**4 timesteps.  This is because the `pr` files have an additional time value for 20 of the years (see table below).  This extra 5 days (20 timesteps / (4 timesteps/day)) of time in the pr files explains the somewhat steadily increasing incoherence.

So it seems like this does not indicate a bug in TECA, though it does motivate consistency-checking in the MCF reader.

It also raises a bigger question: what do we do about this?   Given that there are some other problems with this particular dataset--which is a postprocessed version of CAM output--I'm inclined to avoid using it further.  Daithi has given me access to the original model output files, which won't have this time issue.  Further, the data will be on the model's native vertical grid, so we can recalculate IVT exactly as it would be in the model's state (see #636 ).

Thanks to Yang for pointing out the inconsistency in the number of timesteps.

![Screen Shot 2021-08-09 at 10 43 51 AM](https://user-images.githubusercontent.com/8796694/128725411-23a4bb81-a07b-4028-89aa-9b9576b6ec1d.png)

The table above was generated by the following bash script, which manually went through all of the files in `run001`:
```bash
#!/bin/bash
set -e

# set the variables
VARIABLES="ua va hus pr"

RUN="run001"

# get the list of hus files
input_files=$(ls /global/cscratch1/sd/taobrien/C20C_ua_va_hus/linked_inputs/LBNL/CAM5-1-1degree/All-Hist/est1/v2-0/3hr/atmos/hus/${RUN}/*.nc)

# print the table header
print_str="# YEAR"
for var in $VARIABLES
do
    print_str="${print_str}\t${var}"
done
echo -e ${print_str}

#  loop over hus files
for ffile in $input_files
do

    # initialze the table row with the current year, as parsed from the filename
    print_str=$(python -c "print(\"${ffile}\".split('_')[-1].split('.')[0][:4])")
    
    # loop over variables
    for var in $VARIABLES
    do

        # substitute the current variable name in the current hus file's path
        _tmp=${ffile//hus/$var}
        vfile=${_tmp/C20C_ua_va_${var}/C20C_ua_va_hus}

        # get the number of timesteps, using ncdump
        timestr=$(ncdump -h ${vfile} | grep UNLIMITED)

        # extract the number of timesteps from the string
        timeval=$(python -c "print(\"${timestr}\".split(\"(\")[1].split()[0])")

        # add the number of timesteps to the current row
        print_str="${print_str}\t$timeval"

    done
    # print the row
    echo -e $print_str
done
```

**Comment by burlen** [comment]
Fixed in #639 See https://github.com/LBL-EESA/TECA/pull/639/commits/be48996d7d1456a36e9fc9851f1e4e9a7d38b48c

Running the patch on the above MCF file reports
```
ERROR: [0:46912496728832] [/global/cscratch1/sd/loring/teca_testing/TECA/io/teca_multi_cf_reader.cxx:1752 4.0.0-239-gb2aba11]
ERROR: Time axis missmatch detected on a managed reader. The time axis must be indentical across all managed readers. Correctness cannot be assured. The time axis array "time" from source "3" does not match the reference time axis array "time" from source "0". The arrays have different sizes 169334 and 169354
```

**Comment by taobrienlbl** [comment]
That error message seems perfect!
